### PR TITLE
Improve `fit_all_nuisance` and `fit_all_treatment` docstring

### PR DESCRIPTION
**Pull request by FrancescMartiEscofetQC**
# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Comment by FrancescMartiEscofetQC** [comment]
FYI @ArseniyZvyagintsevQC

File: metalearners/metalearner.py [hunk 1]
```diff
@@ -760,7 +760,18 @@ def fit_all_nuisance(
...
+        is that in the case ``fit_params`` follows the first usage pattern (explained in
+        :meth:`~metalearners.metalearner.MetaLearner.fit`), then the training parameters
+        will only be used for the nuisance models, and in the case they should also be used
+        by the treatment models, these should also be passed in the following call to
+        :meth:`~metalearners.metalearner.MetaLearner.fit_all_treatment`.
+
+        This method, combined with :meth:`~metalearners.metalearner.MetaLearner.fit_all_treatment`,
+        facilitates the segmentation of the metalearner fitting process into two distinct parts.
+        This division allows for interventions between the two stages, such as performing
+        features selection for the treatment models or conducting hyperparameter optimization
```
Comment (kklein) [inline metalearners/metalearner.py:773]:
```suggestion
        feature selection for the treatment models or conducting hyperparameter optimization
```

File: metalearners/metalearner.py [hunk 2]
```diff
@@ -760,7 +760,18 @@ def fit_all_nuisance(
 
         If pre-fitted models were passed at instantiation, these are never refitted.
 
-        For the parameters check :meth:`metalearners.metalearner.MetaLearner.fit`.
+        The only difference with :meth:`~metalearners.metalearner.MetaLearner.fit` parameters,
+        is that in the case ``fit_params`` follows the first usage pattern (explained in
```
Comment (kklein) [inline metalearners/metalearner.py:764]:
```suggestion
        is that if ``fit_params`` follows the first usage pattern (explained in
```

File: metalearners/metalearner.py [hunk 3]
```diff
@@ -777,7 +788,11 @@ def fit_all_treatment(
     ) -> Self:
         """Fit all treatment models of the MetaLearner.
 
-        For the parameters check :meth:`metalearners.metalearner.MetaLearner.fit`.
+        The only difference with :meth:`~metalearners.metalearner.MetaLearner.fit` parameters,
+        is that in the case ``fit_params`` follows the first usage pattern (explained in
```
Comment (kklein) [inline metalearners/metalearner.py:792]:
```suggestion
        is that if ``fit_params`` follows the first usage pattern (explained in
```

**Review by kklein — APPROVED** [review]
Thanks! :)

**Pull request merged**